src/Compilers/CSharp/Test/Semantic/Semantics/ExtensionTypeTests.cs

```diff
@@ -0,0 +1,4993 @@
+            // (3,33): error CS0119: 'UnderlyingClass' is a type, which is not valid in the given context
+            // explicit extension R(int i) for UnderlyingClass { }
+            Diagnostic(ErrorCode.ERR_BadSKunknown, "UnderlyingClass").WithArguments("UnderlyingClass", "type").WithLocation(3, 33),
+            // (3,33): error CS0201: Only assignment, call, increment, decrement, await, and new object expressions can be used as a statement
+            // explicit extension R(int i) for UnderlyingClass { }
+            Diagnostic(ErrorCode.ERR_IllegalStatement, "UnderlyingClass").WithLocation(3, 33),
+            // (3,49): error CS1002: ; expected
+            // explicit extension R(int i) for UnderlyingClass { }
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, "{").WithLocation(3, 49),
+            // (3,49): error CS1525: Invalid expression term '{'
+            // explicit extension R(int i) for UnderlyingClass { }
+            Diagnostic(ErrorCode.ERR_InvalidExprTerm, "{").WithArguments("{").WithLocation(3, 49),
+            // (3,49): error CS1002: ; expected
+            // explicit extension R(int i) for UnderlyingClass { }
+            Diagnostic(ErrorCode.ERR_SemicolonExpected, "{").WithLocation(3, 49),
+            // (3,49): error CS1026: ) expected
+            // explicit extension R(int i) for UnderlyingClass { }
+            Diagnostic(ErrorCode.ERR_CloseParenExpected, "{").WithLocation(3, 49)
+            );
+    }
+
+    [Fact]
+    public void UnderlyingType_InstanceType_StaticExtension()
+    {
+        var src = """
+class UnderlyingClass { }
+static explicit extension R for UnderlyingClass
+{
+}
+""";
+        var comp = CreateCompilation(src);
+        var r = comp.GlobalNamespace.GetTypeMember("R");
+        VerifyExtension<SourceExtensionTypeSymbol>(r);
+        Assert.Equal("UnderlyingClass", r.ExtensionUnderlyingTypeNoUseSiteDiagnostics.ToTestDisplayString());
+        Assert.True(r.IsStatic);
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void UnderlyingType_StaticType_StaticExtension()
+    {
+        var src = """
+static class UnderlyingClass { }
+static explicit extension R for UnderlyingClass
+{
+}
+""";
+        var comp = CreateCompilation(src);
+        var r = comp.GlobalNamespace.GetTypeMember("R");
+        VerifyExtension<SourceExtensionTypeSymbol>(r);
+        Assert.Equal("UnderlyingClass", r.ExtensionUnderlyingTypeNoUseSiteDiagnostics.ToTestDisplayString());
+        Assert.True(r.IsStatic);
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void UnderlyingType_StaticType_InstanceExtension()
+    {
+        var src = """
+static class UnderlyingClass { }
+explicit extension R for UnderlyingClass
+{
+}
+""";
+        var comp = CreateCompilation(src);
+        var r = comp.GlobalNamespace.GetTypeMember("R");
+        VerifyExtension<SourceExtensionTypeSymbol>(r);
```

Comment:
Consider making this a bit more helpful: it could suggest making `R` `static`, for example.

Reply:
The static extension with static underlying type scenario is covered in `UnderlyingType_StaticType_StaticExtension`. Let me know if that was not what you had in mind.

Reply:
I am specifically talking about CS9206. It could suggest making `R` static.

Reply:
I am specifically talking about CS9206. It could suggest making `R` static.